### PR TITLE
PIPELINE-2664 Add validation for all fields in VMS messages and enhance tests

### PR DESCRIPTION
**Pull request by rdgfuentes**
https://globalfishingwatch.atlassian.net/browse/PIPELINE-2664

This pull request introduces enhancements to the validation logic for handling invalid values in `pipe_segment/transform/invalid_values.py` and adds comprehensive test coverage in `tests/transform/test_invalid_values.py`. The changes include new validators for VMS message types, a utility function for validating multiple fields, and corresponding test cases to ensure correctness. Additionally, a minor update was made to the `compose.yaml` file to specify the platform for the BigQuery emulator.

### Validation Enhancements

* **New Record-Level Validation:**
  - Added `validate_all_fields_record` function to check if all specified fields in a record are invalid and set them to `None` if so.
  - Introduced `INVALID_RECORD_RULES_BY_MESSAGE_TYPE` for defining record-level validation rules, such as marking VMS messages with `lat` and `lon` both set to 0 as invalid.

* **Field-Level Validation Updates:**
  - Added new field validators for VMS message types to validate `lat`, `lon`, `course`, and `speed` fields.

* **Unified Validation Pipeline:**
  - Refactored `filter_invalid_values` to apply both field-level and record-level validators using `apply_field_validators` and `apply_record_validators`.

### Test Suite Enhancements

* **Expanded Test Cases:**
  - Added test cases for VMS message validation, covering various scenarios like valid values, boundary values, and invalid values for `lat` and `lon`.
  - Added unit tests for `validate_all_fields_record` and `float_to_fixed_point` functions to ensure correctness.

### Configuration Update

* **Compose File Update:**
  - Specified `platform: linux/x86_64` for the BigQuery emulator service in `compose.yaml` to ensure compatibility with osx.

### CODE REVIEW

**Review by tomaslink — CHANGES_REQUESTED** [review]
LGTM but I made some suggestions that I think can simplify the solution.

File: pipe_segment/transform/invalid_values.py [hunk 1]
```diff
@@ -57,6 +57,35 @@ def validate_fixed_position_field(precision, is_invalid):
     )
 
 
+def validate_all_fields_record(fields, is_invalid):
```
[the remark this answers is not in the document]
Comment (rdgfuentes) [reply pipe_segment/transform/invalid_values.py:60]: The docstrings here follow the same format that we currently have in other functions of the same file.

[the remark this answers is not in the document]
Comment (tomaslink) [reply pipe_segment/transform/invalid_values.py:60]: I know, but is not standard format. Even if we don't change the existing code, at least we can try to properly format the new code.

**Comment by tomaslink** [comment]
@tomaslink.
Mention so that this PR appears in "Your pull requests" section.

File: pipe_segment/transform/invalid_values.py [hunk 2]
```diff
@@ -145,10 +174,36 @@ def validate_fixed_position_field(precision, is_invalid):
...
+        "course": validate_field(lambda x: x < 0 or x >= 360),
+        "speed": validate_field(lambda x: x < 0),
+    },
 }
 
+INVALID_RECORD_RULES_BY_MESSAGE_TYPE = {
+    "VMS": {
+        # VMS messages with both lat and lon set to 0 are invalid
+        # and should be set to None
+        "lat,lon": validate_all_fields_record(["lat", "lon"],
```
[the remark this answers is not in the document]
Comment (rdgfuentes) [reply pipe_segment/transform/invalid_values.py]: Thanks for your suggestions, Made some changes based on your comments, looks cleaner now. However I kept the `validate_all_fields_record` because it is consistent with the validators that are already being applied per field.

File: pipe_segment/transform/invalid_values.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (rdgfuentes) [reply pipe_segment/transform/invalid_values.py]: Reformatted all docstrings in the file

**Review by tomaslink — COMMENTED** [review]
Changes looks good. I left another comment.

**Review by tomaslink — APPROVED** [review]
